models/note.go

```diff
@@ -0,0 +1,52 @@
+package models
+
+import (
+	"encoding/json"
+	"time"
+)
+
+type NoteId int64
+
+type NoteType int
+
+const (
+	MARGINALIA NoteType = iota
+	META
+	QUESTIONS
+	PREDICTIONS
+)
+
+func (noteType NoteType) String() string {
+	types := [...]string{
+		"Marginalia",
+		"Meta",
+		"Questions",
+		"Predictions",
+	}
+
+	if noteType < MARGINALIA || noteType > PREDICTIONS {
+		return "Unknown"
+	}
+
+	return types[noteType]
+}
+
+type Note struct {
+	AuthorId      UserId        `json:"authorId"`
+	Type          NoteType      `json:"type"`
+	Content       string        `json:"content"`
+	PublicationId PublicationId `json:"publicationId"`
+	CreationTime  time.Time     `json:"creationTime"`
+}
+
+func (note *Note) MarshalJSON() ([]byte, error) {
+	type Alias Note
+
+	return json.Marshal(&struct {
```

Comment:
I don't understand what this does. It converts a Note into a []byte? Why doe we need the Alias type?

Reply:
oh yeah ample comments would probably make sense here. It converts it into a byte array bc that's just what the builtin json.Marshall (equivalent to json.dumps in python) returns. The Alias thing was something i found online for custom json parsing. The reason this needed custom parsing was because we need to convert NoteType from being an integer into some readable form for the front end

Reply:
Not sure i understand completely. But it somehow helps to make a custom json serializer?

Reply:
yup. anything i can clarify?

Reply:
Would it still have worked if you didn't create the Alias type? aka find and replaced Alias with Note and removed line 43?  Or is it used so that Golang specifically doesn't use the Type Note marshaling?

Reply:
not sure. I expect converting note to type Note directly instead of indirectly might be problematic? But I really don't know. Can't say I entire understand what I've done. Perhaps we should see if @ethomas2 can explain what going on here

Reply:
Sorry i'm not sure i follow how you would get the side affect of line 49, while getting ride of this method. We added this because it was inserting the Type number instead of calling the String method.